src/main.cc

```diff
@@ -333,170 +330,171 @@ NAN_METHOD(CreateKey)
-    {
-      Nan::ThrowTypeError("RegOpenKeyEx : cannot find the registrykey, error_code : ERROR_FILE_NOT_FOUND");
-      return;
-    }
-    else if (openKey == ERROR_SUCCESS)
-    {
-      long setValue = ERROR_INVALID_HANDLE;
+  HKEY hOpenKey;
+  LONG openKey = RegOpenKeyEx(
+      first,
+      subkey,
+      0,
+      KEY_WRITE | KEY_WOW64_64KEY,
+      &hOpenKey);
 
-      if (wcscmp(valueType, L"REG_SZ") == 0)
-      {
-        Nan::Utf8String typeArg(Nan::To<v8::String>(info[4]).ToLocalChecked());
-        auto valueData = utf8ToWideChar(std::string(*typeArg));
-        if (valueData == nullptr)
-        {
-          Nan::ThrowTypeError("A string was expected for the fifth argument, but could not be parsed.");
-          return;
-        }
-        int datalength = static_cast<int>(wcslen(valueData) * sizeof(valueData[0]));
-        setValue = RegSetValueEx(
-            hOpenKey,
-            valueName,
-            0,
-            REG_SZ,
-            (const BYTE *)valueData,
-            datalength);
-      }
-      else if (wcscmp(valueType, L"REG_DWORD") == 0)
-      {
-        int dwordData = Nan::To<int>(info[4]).FromJust();
-        DWORD valueData = static_cast<DWORD>(dwordData);
-
-        setValue = RegSetValueEx(
-            hOpenKey,
-            valueName,
-            0,
-            REG_DWORD,
-            (const BYTE *)&valueData,
-            sizeof(valueData));
-      }
-      else
-      {
-        char errorMessage[255];
-        sprintf_s(errorMessage, "RegSetValueEx Unmanaged type : '%ls'", valueType);
-        Nan::ThrowTypeError(errorMessage);
-        return;
-      }
+  if (openKey == ERROR_FILE_NOT_FOUND)
+  {
+    Napi::TypeError::New(env, "RegOpenKeyEx : cannot find the registrykey, error_code : ERROR_FILE_NOT_FOUND").ThrowAsJavaScriptException();
+    return env.Undefined();
+  }
+  else if (openKey == ERROR_SUCCESS)
+  {
+    long setValue = ERROR_INVALID_HANDLE;
 
-      if (setValue != ERROR_SUCCESS)
+    if (wcscmp(valueType, L"REG_SZ") == 0)
+    {
+      std::string typeArg = info[4].As<Napi::String>();
+      auto valueData = utf8ToWideChar(typeArg);
+      if (valueData == nullptr)
       {
-        // FIXME: the key does not exist, just return an empty array for now
-        info.GetReturnValue().Set(New<v8::Boolean>(false));
-        return;
+        Napi::TypeError::New(env, "A string was expected for the fifth argument, but could not be parsed.").ThrowAsJavaScriptException();
+        return env.Undefined();
       }
-      info.GetReturnValue().Set(New<v8::Boolean>(true));
-      RegCloseKey(hOpenKey);
+      int datalength = static_cast<int>(wcslen(valueData) * sizeof(valueData[0]));
+      setValue = RegSetValueEx(
+          hOpenKey,
+          valueName,
+          0,
+          REG_SZ,
+          (const BYTE *)valueData,
+          datalength);
+    }
+    else if (wcscmp(valueType, L"REG_DWORD") == 0)
+    {     
+      uint32_t dwordData = info[4].ToNumber().Uint32Value();
```

Comment:
This is the only change I made with respect the original behavior: we were using `int`, but [DWORD is an unsigned integer of 32bits](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-dtyp/262627d8-3418-4627-9218-4ffe110850b2), so using `uint32_t` is much more correct.